src/pages/messages/components/NviCandidatesList.tsx

```diff
@@ -119,22 +118,26 @@ export const NviCandidatesList = ({
           ) : (
             <ListPagination
               count={nviCandidatesQuery.data?.totalHits ?? 0}
-              rowsPerPage={rowsPerPage}
+              rowsPerPage={nviParams.size}
               page={page}
-              onPageChange={(newPage) => setPage(newPage)}
+              onPageChange={(newPage) => {
+                searchParams.set(NviCandidatesSearchParam.Offset, ((newPage - 1) * nviParams.size).toString());
+                history.push({ search: searchParams.toString() });
+              }}
               onRowsPerPageChange={(newRowsPerPage) => {
-                setRowsPerPage(newRowsPerPage);
-                setPage(1);
+                searchParams.set(NviCandidatesSearchParam.Size, newRowsPerPage.toString());
+                searchParams.delete(NviCandidatesSearchParam.Offset);
+                history.push({ search: searchParams.toString() });
               }}
               maxHits={10_000}>
               <List data-testid={dataTestId.tasksPage.nvi.candidatesList} disablePadding sx={{ mb: '0.5rem' }}>
                 {nviCandidatesQuery.data?.hits.map((nviCandidate, index) => {
-                  const currentOffset = (page - 1) * rowsPerPage + index;
+                  const currentOffset = (page - 1) * nviParams.size + index;
                   return (
                     <ErrorBoundary key={nviCandidate.identifier}>
                       <NviCandidateListItem
                         nviCandidate={nviCandidate}
-                        nviQueryParams={nviQueryParams}
+                        nviQueryParams={nviParams}
```

Comment:
Liker dere best at denne sendes ned som en prop, eller at `NviCandidateListItem` selv henter den med `const nviParams = useNviCandidatesParams()`?
Jeg er usikker. Den siste har kanskje litt mer lettlest kode, men vil samtidige lage nøyaktig samme params-objekt flere ganger (en gang for hver item faktisk), så det er jo litt unødvendig prosessering der 🤔

Reply:
Readability og unngåing av props synes jeg kan veie opp for litt prosessering. Er vel ikke snakk om store forskjellen? Stemmer litt for siste, men har ikke noen sterk formening.

Reply:
Stemmer for readability 👍

Reply:
Har oppdatert :)